### PR TITLE
[12.0][ADD] fieldservice_route_stock

### PR DESCRIPTION
**Pull request by nikitavaghela**
* Added field_service_route_stock module as per #383

### CODE REVIEW

File: fieldservice_route_stock/models/fsm_route_dayroute.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        stage_obj = self.env['fsm.stage']
+        accout_move_obj = self.env['account.move']
+        # default inventory of all category so used all category
+        product_categ = self.env.ref('product.product_category_all')
+        for rec in self:
+            journal_id = False
+            amount = 0
+            if vals.get('stage_id'):
+                account = (rec.fsm_vehicle_id.person_id.
+                           property_account_payable_id)
```
Comment (max3903) [inline fieldservice_route_stock/models/fsm_route_dayroute.py]:
```suggestion
                           property_account_receivable_id)
```

File: fieldservice_route_stock/views/fsm_route_dayroute.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            <xpath expr="//sheet/notebook/page[2]" position="after">
+                <page string="Inventory" name="inventory">
+                    <group>
+                        <field name="final_inventory_id" context="{'dayroute_id': id}"/>
+                    </group>
+                </page>
+            </xpath>
+        </field>
+    </record>
+
```
Comment (max3903) [inline fieldservice_route_stock/views/fsm_route_dayroute.xml:17]:
```suggestion
```

File: fieldservice_route_stock/models/stock_inventory.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    dayroute_id = fields.Many2one(
+        'fsm.route.dayroute', string='ay Route')
+
+    adjustment_account_move_id = fields.Many2one('Adjustment Move')
+
+    @api.model
+    def default_get(self, default):
+        res = super(Inventory, self).default_get(default)
+        route_obj = self.env['fsm.route.dayroute']
```
Comment (max3903) [inline fieldservice_route_stock/models/stock_inventory.py]:
```suggestion
```

File: fieldservice_route_stock/models/stock_inventory.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        if self._context.get('dayroute_id'):
```
Comment (max3903) [inline fieldservice_route_stock/models/stock_inventory.py:20]:
```suggestion
        if self._context.get('dayroute_id'):
            route_obj = self.env['fsm.route.dayroute']
```

File: fieldservice_route_stock/models/stock_inventory.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            dayroute = route_obj.browse(self._context.get('dayroute_id'))
+            if dayroute and dayroute.fsm_vehicle_id:
+                res.update(
+                    {'location_id': dayroute.fsm_vehicle_id.
+                     inventory_location_id.id,
+                     'name': dayroute.fsm_vehicle_id.name})
```
Comment (max3903) [inline fieldservice_route_stock/models/stock_inventory.py]: @nikitavaghela Name of the vehicle + the date.

File: fieldservice_route_stock/models/fsm_route_dayroute.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    @api.multi
+    def write(self, vals):
+        stage_obj = self.env['fsm.stage']
+        accout_move_obj = self.env['account.move']
+        # default inventory of all category so used all category
+        product_categ = self.env.ref('product.product_category_all')
+        for rec in self:
+            journal_id = False
+            amount = 0
+            if vals.get('stage_id'):
```
Comment (max3903) [inline fieldservice_route_stock/models/fsm_route_dayroute.py]:
```suggestion
            if vals.get('stage_id', False):
```

**Comment by nikitavaghela** [comment]
@max3903, I added `fieldservice_fleet` as a dependency of this module because in the unit test case when I create a vehicle it will be raised error because of model_id is required in `fieldservice_fleet` module.

model

**Comment by max3903** [comment]
@nikitavaghela Please revert your changes.

@brian10048 Can we set a default value for `model_id` like "Default" or "Unknown"?

File: fieldservice_route_stock/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Field Service Route Stock',
+    'version': '12.0.1.0.0',
+    'category': 'Field Service',
+    'author': 'Open Source Integrators, Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/field-service',
+    'depends': [
+        'account',
```
Comment (max3903) [inline fieldservice_route_stock/__manifest__.py]:
```suggestion
```

File: fieldservice_route_stock/__manifest__.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        'fieldservice_route',
+        'fieldservice_vehicle_stock',
+        'fieldservice_fleet',
```
Comment (max3903) [inline fieldservice_route_stock/__manifest__.py]:
```suggestion
```

File: fieldservice_route_stock/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+{
+    'name': 'Field Service Route Stock',
+    'version': '12.0.1.0.0',
+    'category': 'Field Service',
+    'author': 'Open Source Integrators, Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/field-service',
+    'depends': [
+        'account',
+        'fieldservice_route',
```
Comment (max3903) [inline fieldservice_route_stock/__manifest__.py]:
```suggestion
        'fieldservice_route_vehicle',
```

File: fieldservice_route_stock/__manifest__.py (hunk 7 above)
Comment (nikitavaghela) [reply fieldservice_route_stock/__manifest__.py]: for Jornal entries, we need to add `account` dependency.

**Comment by nikitavaghela** [comment]
> @nikitavaghela Please revert your changes.
> 
> @brian10048 Can we set a default value for `model_id` like "Default" or "Unknown"?

Revert is Done and Travis is green.

File: fieldservice_route_stock/__manifest__.py [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+# Copyright (C) 2019 Serpent Consulting Services
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Field Service Route Stock',
+    'version': '12.0.1.0.0',
+    'category': 'Field Service',
+    'author': 'Open Source Integrators, Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/field-service',
+    'depends': [
```
Comment (max3903) [inline fieldservice_route_stock/__manifest__.py:11]:
```suggestion
    'depends': [
        'stock_account',
```

Comment (nikitavaghela) [reply fieldservice_route_stock/__manifest__.py:11]: @max3903, Updated the dependency.

**Review by max3903 — APPROVED** [review]

File: fieldservice_route_stock/tests/test_fieldservice_route_stock.py [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+        self.worker.property_account_receivable_id = \
+            self.receivable_account_id.id
+        self.worker.property_account_payable_id = self.payable_account_id.id
+        self.default_stage = self.env.ref(
+            'fieldservice_route.fsm_stage_route_new')
+        self.closed_stage = self.env.ref(
+            'fieldservice_route.fsm_stage_route_close')
+        self.model_id = self.env.ref('fleet.model_corsa')
+        # self.vehicle_id  = self.env.ref('fleet.vehicle_1')
```
Comment (murtuzasaleh) [inline fieldservice_route_stock/tests/test_fieldservice_route_stock.py]: Please remove it if it's not needed.

**Comment by murtuzasaleh** [comment]
Code Review LGTM.

**Review by murtuzasaleh — APPROVED** [review]

**Pull request merged**